core/src/test/scala/unit/kafka/server/BrokerLifecycleManagerTest.scala

```diff
@@ -197,11 +197,14 @@ class BrokerLifecycleManagerTest {
     result
   }
 
-  def poll[T](context: RegistrationTestContext, manager: BrokerLifecycleManager, future: Future[T]): T = {
-    while (!future.isDone || context.mockClient.hasInFlightRequests) {
-      context.poll()
+  def poll[T](ctx: RegistrationTestContext, manager: BrokerLifecycleManager, future: Future[T]): T = {
+    while (ctx.mockChannelManager.unsentQueue.isEmpty) {
+      if (manager.eventQueue.isEmpty)
```

Comment:
Why do we need this check? If the eventQueue has a event scheduled event at a future time, we need to advance the time to drain the event, right?

Reply:
You're right, this is incorrect. We must only advance the time if the eventQueue has an event scheduled at a future time, unless the next event is a timeout event.